### ISSUE TITLE
Cumulative Time Table

### ISSUE THREAD

**Issue by bradleysigma** [jeroenheijmans/advent-of-code-charts#48]
Is it possible to add a table like the following, showing the total time taken to solve part two of each day (`+`s show how many days have not been completed)?

```
Lexie                                      7 hrs  8 mins 47 secs
Haseeb                                    20 hrs 16 mins 43 secs
Alayah                             1 day   4 hrs 18 mins 21 secs
Anita                              1 day   6 hrs 49 mins 45 secs +
Mila-Rose                          1 day  13 hrs 33 mins 18 secs +
Karter                             2 days 14 hrs 30 mins 22 secs ++
Ruairi                             5 days 13 hrs 47 mins 38 secs
Haroon                             9 days 17 hrs 30 mins 17 secs +++
Abdurrahman                       15 days 14 hrs  6 mins 47 secs +++++
Qasim                             15 days 18 hrs  6 mins  6 secs ++++++
```

(This data comes from my own leaderboard, with the names changed.)

**Comment by jeroenheijmans** [comment]
That does sound like fun! It's not too easy to understand as a feature at first sight though, so I'd have to consider if it's worth the screen real estate...

I'll leave it open for now, and might consider it next year November(ish) if I have the time. Sounds reasonable?

**Comment by jeroenheijmans** [comment]
Just a quick update, I don't think I myself will work on such a graph in 2021, got enough other things going on this year already.

If others would want to take a stab at this do ping me here so we can have a chat, I'm still not entirely sure if/how it would be a good addition (though the idea's interesting).

**Comment by jeroenheijmans** [comment]
I'm working on a few updates this year, and gave this suggestion some more thought.

I still like the idea of something like this, but couldn't come to a solution where this table would function in a useful manner. Most importantly, if there's a day (or two) where you complete it only days later: it messes up the entire ranking.

For example, I might be fast on my leaderboard, completing most days within 3 hours or less. But then there might be 1 or 2 days each year where I can't figure them out, let those puzzles sit for days on end, until I have more time (e.g. on a weekend) to finish them up.

The https://adventofcode.com/2021/leaderboard/self statistics kindof solve this by saying just "`>24h`" for days that took _way_ too long. But even if I start it in the morning, go to work, and finish up in the evening, a completion time of 14+ hours is possible.

Perhaps something where only your "best 15 out of 25 days" are counted would work for most leaderboards? That also feels iffy though.

In short, I'm not sure yet. I'm open to suggestions, but for now going to let this issue sit a bit longer.

**Comment by ogoletti** [comment]
Maybe a solution could be to have a different rating for delta time. 
Then it's just another tab in the "Stats Per Day: 1 2 ... Cumulative" where players of a leaderboard are sorted on average ranking throughout passed days. The actual total time wouldn't be important but just the ranking.

**Comment by jeroenheijmans** [comment]
I did choose to go with @ogoletti's suggestion, partially because #58 by @gg-dt contained the basis for just that feature. I think when those changes land (via #79, for 2023) it'll solve the largest part of the use case from OP (I hope?) with a visual like this:

![image](https://user-images.githubusercontent.com/1590536/211115803-39f2c658-ecf0-48bb-8621-dd8a05b210a7.png)

True, you don't see the exact feature requested by OP (adding up all the delta times), but I don't think that's super useful in practice anyways: there will often be a day where you get stuck on part 2 and solve it _daaaaayyyyss_ later. Thus the cumulative value doesn't mean much.

The table from #58 has similar problems but also offered ways to solve that:

- The primary sort in that table is based on "points" earned per day, regardless of how _big_ your delta was;
- The mean/median exclude deltas over 4 hours;

Thanks a bunch for the feature request, the input, and the further discussion, all! I hope the chosen path is suitable (enough).
